### PR TITLE
`EmptyNodeCapabilities` for chains that don't discriminate nodes

### PR DESCRIPTION
**Pull request by neysofu**
Provides a default `NodeCapabilities` implementor called `EmptyNodeCapabilities`, so that new chains can use that without writing their own, assuming they don't need any custom logic.

Also "fixes" the `PartialOrd` implementation of Ethereum's `NodeCapabilities`, which is not buggy per se but breaks the required trait invariants.

Related to https://github.com/graphprotocol/graph-node/issues/3260.

### CODE REVIEW

**Comment by neysofu** [comment]
Alright, I think I see how indeed we can remove all traces of `NodeCapabilities` from the `Blockchain` trait. Some problems arise with `BlockStreamBuilder::build_firehose` though, where the current flow of logic is `TriggerFilter -> NodeCapabilities -> TriggersAdapter` via `Blockchain::triggers_adapter`. This can be transformed into `TriggerFilter -> &[Chain::DataSource] -> TriggersAdapter`, but getting the data sources out of a `TriggerFilter` is not immediate and it would require refactoring that trait as well. I'd leave it for a future PR, unless I'm missing something and it's easier than that.

**Comment by leoyvens** [comment]
Afaict `fn build_firehose` does not have a good reason to be constructing a triggers adapter. I only see it used here 
https://github.com/graphprotocol/graph-node/blob/e5198128bd3d10c95325580ade063f658709fbe4/chain/ethereum/src/chain.rs#L627-L631

That TODO isn't hard to fix, this block needs to be extracted to a new `fn parse_triggers` and that should be used directly instead of going through the triggers adapter
https://github.com/graphprotocol/graph-node/blob/e5198128bd3d10c95325580ade063f658709fbe4/chain/ethereum/src/chain.rs#L529-L536

**Comment by neysofu** [comment]
I had totally forgot about this. I never pursued the follow-up refactoring, should we merge it as it is (after fixing conflicts)?

File: chain/ethereum/src/capabilities.rs [hunk 1]
```diff
@@ -15,28 +13,28 @@ pub struct NodeCapabilities {
...
             self.traces.cmp(&other.traces),
-        ) {
-            (Ordering::Greater, Ordering::Greater) => Ordering::Greater,
-            (Ordering::Greater, Ordering::Equal) => Ordering::Greater,
-            (Ordering::Equal, Ordering::Greater) => Ordering::Greater,
-            (Ordering::Equal, Ordering::Equal) => Ordering::Equal,
-            (Ordering::Less, _) => Ordering::Less,
-            (_, Ordering::Less) => Ordering::Less,
-        }
+        ])
```
Comment (leoyvens) [inline chain/ethereum/src/capabilities.rs:23]: Isn't this just `(&self.archive, &self.traces).cmp((&other.archive, &other.traces))`.

**Review by leoyvens — APPROVED** [review]

File: chain/ethereum/src/capabilities.rs (hunk 1 above)
Comment (neysofu) [reply chain/ethereum/src/capabilities.rs:23]: No, Rust tuples define lexicographic ordering, which is not what we use for node capabilities comparisons (e.g. all fields have to be `>=` in order for a `NodeCapabilities` to be `>`, but if we just compare them as tuples, the comparison might stop at `archive` and discard `traces`).

Comment (leoyvens) [reply chain/ethereum/src/capabilities.rs:23]: I see, thanks

**Pull request merged**